src/Ninject/Planning/Targets/Target.cs

```diff
@@ -55,8 +55,8 @@ protected Target(MemberInfo member, T site)
             this.Member = member;
             this.Site = site;
 
-            this.constraint = new Lazy<Func<IBindingMetadata, bool>>(this.ReadConstraintFromTarget);
-            this.isOptional = new Lazy<bool>(this.ReadOptionalFromTarget);
+            this.constraint = new Lazy<Func<IBindingMetadata, bool>>(() => this.ReadConstraintFromTarget());
```

Comment:
@drieseng Can you please help me understand why this change could improve the performance?

Reply:
This is related to the use of method group vs. delegate.
A delegate is cached by the compiler, method group is not.
I'll update this PR (for Array.Empty<T>()) and re-run the benchmarks again to make sure the results are "stable".

Reply:
Thanks!

Reply:
I verified that the (minor) performance improvement for the constructor is indeed from replacing a method group by a delegate.